### PR TITLE
add `invalid-rendered-yaml` linter

### PR DESCRIPTION
**Pull request by pavansokkenagaraj**
feature:
- add a new linter `invalid-rendered-yaml` which validates the rendered manifests are valid yamls
```json
{
    "lintExpressions": [
        {
            "rule": "invalid-rendered-yaml",
            "type": "error",
            "message": "yaml: (          defaultCliVersion: ***HIDDEN***): did not find expected alphabetic or numeric character",
            "path": "helm/helm-chart.yaml",
            "positions": null
        }
    ],
    "isLintingComplete": false
}
```

chore:
- add okteto debugging steps
- update kots go mod
- remove duplicate imports `goyaml "gopkg.in/yaml.v2"` and use `"gopkg.in/yaml.v2"` as standard

### CODE REVIEW

File: pkg/kots/lint.go [hunk 1]
```diff
@@ -820,6 +824,35 @@ func lintFileHasValidYAML(file SpecFile) []LintExpression {
...
+		var doc interface{}
+		err := yaml.Unmarshal([]byte(renderedFile.Content), &doc)
+		if err != nil {
+			lintErrMsg := err.Error()
+			errLine, err := util.TryGetLineNumberFromValue(err.Error())
+			if err == nil && errLine > -1 {
+				lines := strings.Split(renderedFile.Content, "\n")
+				if len(lines) > errLine {
+					errLineContent := lines[errLine-1]
+					lintErrMsg = strings.Replace(lintErrMsg, fmt.Sprintf("line %d", errLine), fmt.Sprintf("(%s)", errLineContent), 1)
```
Comment (cbodonnell) [inline pkg/kots/lint.go:839]: Could we change this to read with the error first and then the line with the invalid syntax? example:
```
yaml: found character that cannot start any token: alb.ingress.kubernetes.io/inbound-cidrs: ***HIDDEN***/0
```

File: pkg/kots/lint_test.go [hunk 2]
```diff
@@ -4767,3 +4769,109 @@ spec:
...
+						Content: `apiVersion: ***HIDDEN***/0`,
+					},
+				},
+			},
+			want: []LintExpression{
+				{
+					Rule:    "invalid-rendered-yaml",
+					Type:    "error",
+					Path:    "invalid.yaml",
+					Message: "yaml: did not find expected alphabetic or numeric character",
```
Comment (cbodonnell) [inline pkg/kots/lint_test.go:4811]: why does this one not return the line that is invalid in the message? is it an issue with it being in line 1?

**Review by cbodonnell — COMMENTED** [review]
what would happen with a `kots.io/v1beta1` kind `Config`? it looks like it will not be rendered and returned as-is here:
https://github.com/replicatedhq/kots-lint/blob/main/pkg/kots/render.go#L125
https://github.com/replicatedhq/kots-lint/blob/main/pkg/kots/render.go#L64

File: pkg/kots/lint.go (hunk 1 above)
Comment (pavansokkenagaraj) [reply pkg/kots/lint.go]: addressed in 
https://github.com/replicatedhq/kots-lint/pull/159/commits/5bedadc630ecbe1e385b0c5fe0da1dfea8f32051#diff-1d1c16d1fd801c249928ba51957dc01e3720d412f68cb52bbb3cf7f29912122bR838-R841

File: pkg/kots/lint_test.go (hunk 2 above)
Comment (pavansokkenagaraj) [reply pkg/kots/lint_test.go:4811]: the module's `yaml.unmarshall` returns the error in that way. so, wanted to add the test with edge case[in future if module code error handling changes, we can update our code accordingly]

Comment (cbodonnell) [reply pkg/kots/lint_test.go:4811]: gotcha, so yaml library does not return a line number.  this is at least better than nothing, so i'm ok with it

**Review by cbodonnell — APPROVED** [review]
